### ISSUE TITLE
Apple Pay option is not visible with Drop In Components options...

### ISSUE THREAD

**Issue by sumitkapoorshineweb** [Adyen/adyen-ios#372]
I have Adyen in my iOS app, but getting issue with Apple Pay option to show on UI. I am getting Apple Pay option in response from api but as payment methods passed to 'DropInComponent' with configuration. Apple Pay option is not visible in payment methods list...

Can you plz suggest what's possibility of this issue coming or any suggestion that I can try to resolve this issue..

Here is my code to show Payment options:

     let configuration = DropInComponent.PaymentMethodsConfiguration()
    
        configuration.environment = environment
        configuration.localizationParameters = nil
        
        configuration.clientKey = payment.testClientKey
        
        configuration.applePay.merchantIdentifier = "******"

        configuration.applePay.summaryItems = applePaySummaryItems
        
        let dropInComponent = DropInComponent(paymentMethods: paymentMethods,
                                              paymentMethodsConfiguration: configuration)
        
        dropInComponent.delegate = self
        dropInComponent.environment = environment
        dropInComponent.payment = Adyen.Payment(amount: Adyen.Payment.Amount(value: Decimal(totalPrice), currencyCode: "Kr"))
        currentComponent = dropInComponent
        dropInComponent.viewController.navigationItem.leftBarButtonItem = .init(barButtonSystemItem: .cancel,
                                                                              target: self,
                                                                              action: #selector(cancelDidPress))
        
        self.present(dropInComponent.viewController, animated: true)

**Comment by descorp** [comment]
Hi @sumitkapoorshineweb 

What version of SDK are you using and what version of Adyen API?

You can turn on debug logs to see what is happening. 
To do so, in AppDelegate add:

```swift
        #if DEBUG
            AdyenLogging.isEnabled = true
        #endif
```

In general, most common reasons for ApplePay to stay hidden:

1. No ApplePay [capability added to the target](https://help.apple.com/xcode/mac/current/#/deva43983eb7) with selected `MerchantID`.
2. Your device/Appple account does not support ApplePay. You can check [here](https://applepaydemo.apple.com/)
3. Some invalid parameter:  see [list of errors](https://github.com/Adyen/adyen-ios/blob/develop/AdyenComponents/Apple%20Pay/ApplePayComponent.swift#L205-L219)
4. You device have no payment networks that are configured in your Adyen account. For example, you have MasterCard and Visa cards, but your account only configured for Discovery and Amex. Check `brands` collection inside `paymentMethods` object.

**Comment by sumitkapoorshineweb** [comment]
I'm trying to debug with enabling AdyenLogging, but nothing shows in logs...

In Payment Methods, I am getting this :

PaymentMethods(regular: [Adyen.CardPaymentMethod(type: "scheme", name: "Credit Card", fundingSource: nil, brands: ["visa", "mc", "amex"]), Adyen.ApplePayPaymentMethod(type: "applepay", name: "Apple Pay"), Adyen.RedirectPaymentMethod(type: "klarna", name: "Pay later with Klarna."), Adyen.RedirectPaymentMethod(type: "klarna_account", name: "Pay over time with Klarna."), Adyen.RedirectPaymentMethod(type: "paysafecard", name: "Paysafecard")], stored: [])

But with Drop In Components, not showing all above payment methods in list, And in case of Apple Pay -> getting this error 

  case .userCannotMakePayment:
          return "The user can’t make payments on any of the payment request’s supported networks."

And Finally only these methods showing:

1. Credit Card
2. Pay later with Klarna.
3. Pay over time with Klarna.
4. Paysafecard

Plz suggest in which this error coming for Apple Pay...

**Comment by descorp** [comment]
Hey @sumitkapoorshineweb 

What version of Adyen API are you using?

> The user can’t make payments on any of the payment request’s supported networks.

This is the case when your device's AppleWallet have no payment networks that are configured in your Adyen account.
What types of cards are attached to the AppleWallet on your test device?
What `brands` does contain `ApplePayPaymentMethod` if any? Could you print out ApplePayPaymentMethod or `paymentMetho` json?

**Comment by sumitkapoorshineweb** [comment]
I got these debug information for you from Apple Pay Component Class init method -> :

Here is Method :

public init(paymentMethod: ApplePayPaymentMethod,
                payment: Payment,
                configuration: Configuration,
                cancelHandler: (() -> Void)? = nil) 

(lldb) po print(paymentMethod)
ApplePayPaymentMethod(type: "applepay", name: "Apple Pay")
0 elements

(lldb) po print(payment)
Payment(amount: Adyen.Payment.Amount(value: 52700, currencyCode: "Kr"), countryCode: nil)
0 elements

(lldb) po print(configuration)
Configuration(summaryItems: [<PKPaymentSummaryItem: 0x283c8aaf0>, <PKPaymentSummaryItem: 0x283c8a5b0>, <PKPaymentSummaryItem: 0x283c8a550>, <PKPaymentSummaryItem: 0x283c8ad60>, <PKPaymentSummaryItem: 0x283c8a580>, <PKPaymentSummaryItem: 0x283c8a4c0>, <PKPaymentSummaryItem: 0x283c8a430>], merchantIdentifier: "merchant.com.adyen.BroadkaztTechnology.test", requiredBillingContactFields: Set([]), requiredShippingContactFields: Set([]), excludedCardNetworks: [])
0 elements

(lldb) po print(configuration.supportedNetworks)
[__C.PKPaymentNetwork(_rawValue: Visa), __C.PKPaymentNetwork(_rawValue: MasterCard), __C.PKPaymentNetwork(_rawValue: AmEx), __C.PKPaymentNetwork(_rawValue: Discover), __C.PKPaymentNetwork(_rawValue: Interac), __C.PKPaymentNetwork(_rawValue: Maestro), __C.PKPaymentNetwork(_rawValue: JCB)]

And how can I get Adyen API Version from app side. Can you plz let me know?

**Comment by descorp** [comment]
> And how can I get Adyen API Version from app side. Can you plz let me know?

You can't get it from the app. Your backend is calling Adyen API.
No worries, it is not needed any more.

> (lldb) po print(configuration.supportedNetworks)
[__C.PKPaymentNetwork(_rawValue: Visa), __C.PKPaymentNetwork(_rawValue: MasterCard), __C.PKPaymentNetwork(_rawValue: AmEx), __C.PKPaymentNetwork(_rawValue: Discover), __C.PKPaymentNetwork(_rawValue: Interac), __C.PKPaymentNetwork(_rawValue: Maestro), __C.PKPaymentNetwork(_rawValue: JCB)]

This is good. You have most of the cards configured.

What cards do you have on your device's AppleWallet?

**Comment by sumitkapoorshineweb** [comment]
Its Bank of America, Business Debit  Visa Card, But Apple Pay option still not visible

**Comment by descorp** [comment]
hm.. intersting..
Are those the payment methods do you see if you open [Apple Demo](https://applepaydemo.apple.com/) ?

**Comment by sumitkapoorshineweb** [comment]
Thanks @descorp for all the help, there was some issue with user on Adyen I guess, I added a new user on Adyen under separate merchant and it works, still I have setup Currency and country static just for testing to show Apple Pay option, How can I pass dynamic currency and country code? So it works for all the countries where Apple Pay supported...

**Comment by descorp** [comment]
Ok, nice to know that issue was resolved!
If the user situation concerns you - I would advise checking with [Adyen Support](https://www.adyen.help/hc/en-us)
They should help you configure the profile or create a ticket if this is a bug on our portal.

If you don't mind - I have created another topic for this [here](https://github.com/Adyen/adyen-ios/issues/373)
It would be easier for other users and us to follow up on this conversation in the future.

Feel free to close it or continue in this thread with any additional feedback about the user issue.
